src/measure_theory/lebesgue_measure.lean

```diff
@@ -378,6 +378,106 @@ begin
+  (hf : measurable f) (hg : measurable g) (hs : is_measurable s) :
+  μ.prod volume (region_between f g s) = ∫⁻ y in s, ennreal.of_real ((g - f) y) ∂μ :=
+begin
+  rw measure.prod_apply,
+  { have h : (λ x, volume {a | x ∈ s ∧ a ∈ Ioo (f x) (g x)})
+            = s.indicator (λ x, ennreal.of_real (g x - f x)),
+    { funext x,
+      rw indicator_apply,
+      split_ifs,
+      { have hx : {a | x ∈ s ∧ a ∈ Ioo (f x) (g x)} = Ioo (f x) (g x) := by simp [h, Ioo],
+        simp only [hx, real.volume_Ioo, sub_zero] },
+      { have hx : {a | x ∈ s ∧ a ∈ Ioo (f x) (g x)} = ∅ := by simp [h],
+        simp only [hx, measure_empty] } },
+    dsimp only [region_between, preimage_set_of_eq],
+    rw [h, lintegral_indicator];
+    simp only [hs, pi.sub_apply] },
+  { exact is_measurable_region_between hf hg hs },
+end
+
+/-- The volume of the region between two almost everywhere measurable functions on a measurable set
+    can be respresented as a Lebesgue integral. -/
+theorem volume_region_between_eq_lintegral [sigma_finite μ]
+  (hf : ae_measurable f (μ.restrict s)) (hg : ae_measurable g (μ.restrict s))
+  (hs : is_measurable s) :
+  μ.prod volume (region_between f g s) = ∫⁻ y in s, ennreal.of_real ((g - f) y) ∂μ :=
+begin
+  have h₁ : (λ y, ennreal.of_real ((g - f) y))
+          =ᵐ[μ.restrict s]
+              λ y, ennreal.of_real ((ae_measurable.mk g hg - ae_measurable.mk f hf) y) :=
+    eventually_eq.fun_comp (eventually_eq.sub hg.ae_eq_mk hf.ae_eq_mk) _,
+  have h₂ : (μ.restrict s).prod volume (region_between f g s) =
+    (μ.restrict s).prod volume (region_between (ae_measurable.mk f hf) (ae_measurable.mk g hg) s),
+  { apply measure_congr,
+    apply eventually_eq.inter, { refl },
+    exact eventually_eq.inter
+            (eventually_eq.comp₂ (ae_eq_comp' measurable_fst hf.ae_eq_mk
+              measure.prod_fst_absolutely_continuous) _ eventually_eq.rfl)
+            (eventually_eq.comp₂ eventually_eq.rfl _
+              (ae_eq_comp' measurable_fst hg.ae_eq_mk measure.prod_fst_absolutely_continuous)) },
+  rw [lintegral_congr_ae h₁,
+      ← volume_region_between_eq_lintegral' hf.measurable_mk hg.measurable_mk hs],
+  convert h₂ using 1,
+  { rw measure.restrict_prod_eq_prod_univ,
+    exacts [measure.eq_restrict_of_subset_of_measurable (hs.prod is_measurable.univ)
+      (region_between_subset f g s), hs] },
+  { rw measure.restrict_prod_eq_prod_univ,
+    exacts [measure.eq_restrict_of_subset_of_measurable (hs.prod is_measurable.univ)
+      (region_between_subset (ae_measurable.mk f hf) (ae_measurable.mk g hg) s), hs] },
+end
+
+/-- If two functions are integrable on a measurable set, and one function is less
+    than or equal to the other everywhere on that set, then the volume of the region between the
+    two functions can be respresented as an integral. -/
+theorem volume_region_between_eq_integral [sigma_finite μ]
+  (hf : integrable_on f s μ) (hg : integrable_on g s μ)
+  (hs : is_measurable s) (hfg : ∀ x ∈ s, f x ≤ g x) :
```

Comment:
Can you replace `hfg` with an assumption requiring this almost everywhere for `\mu.restrict s`?

Reply:
I just pushed a new version of this proof with this weaker assumption you asked for. I also tried to address all your other recent comments about fixing docstrings, rearranging lemma statements, etc.